cosmos/profiles/base.py

```diff
@@ -24,6 +25,66 @@
 logger = get_logger(__name__)
 
 
+@dataclasses.dataclass
+class DbtConfigVars:
+    send_anonymous_usage_stats: Optional[bool] = False
+    partial_parse: Optional[bool] = None
+    use_experimental_parser: Optional[bool] = None
+    static_parser: Optional[bool] = None
+    printer_width: Optional[int] = None
+    write_json: Optional[bool] = None
+    warn_error: Optional[str] = None
+    warn_error_options: Optional[dict[str, Any]] = None
+    log_format: Optional[str] = None
+    debug: Optional[bool] = None
+    version_check: Optional[bool] = None
+
+    def _validate_data(self) -> None:
+        checks: dict[str, dict[str, Any]] = {
+            "send_anonymous_usage_stats": {"var_type": bool},
+            "partial_parse": {"var_type": bool},
+            "use_experimental_parser": {"var_type": bool},
+            "static_parser": {"var_type": bool},
+            "printer_width": {"var_type": int},
+            "write_json": {"var_type": bool},
+            "warn_error": {"var_type": str},
+            "warn_error_options": {"var_type": dict, "accepted_values": {"include", "exclude"}},
+            "log_format": {"var_type": str, "accepted_values": {"text", "json", "default"}},
+            "debug": {"var_type": bool},
+            "version_check": {"var_type": bool},
+        }
+
+        for field_name, field_def in self.__dataclass_fields__.items():
+            field_value = getattr(self, field_name)
+
+            if not field_value is None:
+                vars_check = checks.get(field_name, {})
+                accepted_values = vars_check.get("accepted_values")
+                var_type = vars_check.get("var_type", Any)
+
+                if not isinstance(field_value, var_type):
+                    raise CosmosValueError(f"dbt config var {field_name}: {field_value} must be a {var_type}")
+
+                if accepted_values:
+                    if field_value not in accepted_values:
+                        raise CosmosValueError(
+                            f"dbt config var {field_name}: {field_value} must be one of {accepted_values}"
+                        )
```

Comment:
You could instead use [pydantic dataclasses](https://docs.pydantic.dev/1.10/usage/dataclasses/) so you could have the type hints be used in the validation itself so you wouldn't need the dictionary that repeats the dataclass fields. e.g. something like the below with the accepted types you're currently using:
```python
from typing import Literal
from pydantic.dataclasses import dataclass

@dataclass
class DbtConfigVars:
    send_anonymous_usage_stats: bool | None = False
    warn_error_options: Literal["include", "exclude"] | None = None
    log_format: str | None = None
...
```
Then you wouldn't even need the post init and `_validate_data` methods.

Reply:
airflow have version of pydantic. It depend on airflow version. Will check diff

Reply:
fix)

Reply:
have problem in python 3.8

Reply:
What was the issue? Can you point to the test that was failing for Python 3.8. Sometimes with 3.8 the failures are from type hint annotations like `some_arg: str | None`  that can be fixed by added `from __future__ import annotations`.

Reply:
Fix, thanks for help)
the main problem is that I did it at night)

Reply:
That's great it looks much cleaner using pydantic dataclasses for validation!